### ISSUE TITLE
[Cloudflare] ERROR: [CLIENT API] Array

### ISSUE THREAD

**Issue by janvitos** [cloudflare/Cloudflare-WordPress#266]
Hi,

I noticed that Cloudflare now throws an error in the nginx error log whenever I publish or update a post. 

Here's the exact error log:

```
2020/07/01 11:43:45 [error] 3849136#3849136: *1285712 FastCGI sent in stderr: "PHP message: [Cloudflare] ERROR: [CLIENT API] Array
(
    [type] => request
    [path] => (hidden)
    [timeout] => 30
    [method] => DELETE
    [headers] => Array
        (
            [Content-Type] => application/json
            [X-Auth-Email] => (hidden)
            [X-Auth-Key] => (hidden)
        )

    [body] => {"files":["https:\/\/www.hidden.com\/sujet\/actualite\/","https:\/\/www.hidden.com\/tag\/bizarre\/","https:\/\/www.hidden.com\/tag\/catastrophe\/","https:\/\/www.hidden.com\/tag\/france\/","https:\/\/www.hidden.com\/tag\/humour\/","https:\/\/www.hidden.com\/tag\/mignon\/","https:\/\/www.hidden.com\/tag\/monde\/","https:\/\/www.hidden.com\/tag\/sante\/","https:\/\/www.hidden.com\/tag\/societe\/","https:\/\/www.hidden.com\/auteur\/hidden\/","https:\/\/www.hidden.com\/auteur\/hidden\/feed\/","https:\/\/www.hidden.com","https:\/\/www.hidden.com?feed=rss2","https:\/\/...PHP message: [Cloudflare] ERROR: [CLIENT API] Array
(
    [type] => response
    [reason] => Bad Request
    [code] => 400
)
 PHP message: [Cloudflare] ERROR: Bad Request PHP message: [Cloudflare] ERROR: [CLIENT API] Array
(
    [type] => request
    [path] => (hidden)
    [timeout] => 30
    [method] => DELETE
    [headers] => Array
        (
            [Content-Type] => application/json
            [X-Auth-Email] => (hidden)
            [X-Auth-Key] => (hidden)
        )

    [body] => {"files":["https:\/\/www.hidden.com\/sujet\/actualite\/","https:\/\/www.hidden.com\/tag\/bizarre\/","https:\/\/www.hidden.com\/tag\/catastrophe\/","https:\/\/www.hidden.com\/tag\/france\/","https:\/\/www.hidden.com\/tag\/humour\/","https:\/\/www.hidden.com\/tag\/mignon\/","https:\/\/www.hidden.com\/tag\/monde\/","https:\/\/www.hidden.com
```

I have removed my email, website name and auth key from the logs, but I know the values are correct.

Thanks.

**Comment by janvitos** [comment]
I have disabled Automatic Cache Management for now and am using my own code to clear the CloudFlare cache.

**Comment by olegcarguy** [comment]
I am seeing this too

**Comment by gramorris** [comment]
To add to this....

On the 30th June CF introduced a 30 url limit to purge requests (https://api.cloudflare.com/#zone-purge-files-by-url). Currently the plugin makes no provision for that limit and by and large fails relatively silently. We've put a temporary fix on our sites by removing http urls using the purge filter but in a https site it is relatively easy to hit the 30 purge limit if a post contains 5 or so terms.

**Comment by patryk** [comment]
Thanks for patience. There was an unfortunate coincidence of two things:
- Extended number of URLs purged on content update https://github.com/cloudflare/Cloudflare-WordPress/pull/208
- Cache team separately reduced number of allowed URLs in single Cache Purge request

Release 3.5.1 contains the fix. Apologies for the trouble! 🙇‍♂️

**Comment by mnlgmz** [comment]
Hi,

Thank you so much @gramorris for those details.

What I did to fix this issue is removing or commenting from /public/wp-content/plugins/cloudflare/src/WordPress/Hooks.php on line 236, the following:

```
// Purge https and http URLs (Line 235)
//if (function_exists('force_ssl_admin') && force_ssl_admin()) {
//    $listofurls = array_merge($listofurls, str_replace('https://', 'http://', $listofurls));
//} elseif (!is_ssl() && function_exists('force_ssl_content') && force_ssl_content()) {
//    $listofurls = array_merge($listofurls, str_replace('http://', 'https://', $listofurls));
//}
```

**NOTE:**

- If the site is using **force_ssl_admin**, **force_ssl_content**, or a plugin to redirect **HTTP to HTTPS**, you probably will need this.

- If the site is using native HTTPS, you won’t need this at all.

**Feeds (Optional):**

By default, WordPress comes with some feed types, but the most commonly used is RSS2; therefore, if you are not using RDF, RSS (1.0), or Atom feeds, you can disable them.

```
// Feeds (Line 217)
array_push(
    $listofurls,
    //get_bloginfo_rss('rdf_url'),
    //get_bloginfo_rss('rss_url'),
    get_bloginfo_rss('rss2_url'),
    //get_bloginfo_rss('atom_url'),
    get_bloginfo_rss('comments_rss2_url'),
    get_post_comments_feed_link($postId)
);
```

If you are using a third-party comments plugin like Disqus or Spot.IM, you can also disable **comments_rss2_url**, and also **post_comments_feed_link**, which is used to track the comments on a specific post.

```
// Feeds (Line 217)
array_push(
    $listofurls,
    get_bloginfo_rss('rss2_url')
);
```
Hope that helps,

Cheers.

**Comment by janvitos** [comment]
I can confirm that the issue is fixed in 3.5.1.

Thanks for the speedy fix!
